R/utils.R

```diff
@@ -526,6 +585,45 @@ getPrecursorSubset <- function(precursors, params){
   c(pepStart, pepEnd)
 }
 
+#' Re-Assign FDR Aligned Peaks
+#'
+#' This method will re-assign peaks with either the MS2 FDR if there was not enough confidence (identifying transitions) in the IPF FDR, or it will assign the user defined alignedFDR. This assumes the reference run has a really good peak that meets the IPF FDR threshold.
+#'
+#' @author Justin Sing, \email{justinc.sing@mail.utoronto.ca}
+#'
+#' ORCID: 0000-0003-0386-0092
+#'
+#' License: (c) Author (2020) + GPL-3
+#' Date: 2020-07-17
+#' @param dt (data.table) Aligned results tables from writeTables
+#' @param refRuns (dataframe) A peptide reference run table from getRefRun
+#' @param fileInfo (dataframe) A table with run filename information from getRunNames
+#' @param params (list) A list of parameters used by DIAlignR generated from paramsDIAlignR
+#' @return (data.table) Aligned results table with reassigned m-scores
+#' @keywords internal
+#' @seealso writeTables, getRefRun, getRunNames, paramsDIAlignR
+#' @importFrom dplyr recode select
+#' @import data.table
+#' @examples
+#' \dontrun{
+#' finalTbl <- ipfReassignFDR(finalTbl, refRuns, fileInfo, params)
+#' }
+ipfReassignFDR <- function(dt, refRuns, fileInfo, params){
+  ## Create Reference Run table with runName to map back to dt
+  runNameMap <- fileInfo$runName
+  names(runNameMap) <- rownames(fileInfo)
+  refRuns$ref_run <- dplyr::recode(refRuns$run, !!!runNameMap )
```

Comment:
I suggest not to use dplyr because if the table is big 4-5 Gb, it will be really slow

Reply:
Make sense, resolved: https://github.com/shubham1637/DIAlignR/pull/15/commits/21c3b513bbbff86d55a1fdfb466daa4fab948f72